aws/ci_build.rake

```diff
@@ -176,18 +176,12 @@ file deploy_dir('rebuild') do
   touch deploy_dir('rebuild')
 end
 
-def stop_frontend(name, host, log_path)
-  # NOTE: The order of these prefers deploy-speed over user-experience. Stopping varnish first
-  #   immediately terminates connections to the backends so they stop immediately. Changing the
-  #   order to stop Dashboard and Pegasus first would drain the user connections before stopping
-  #   which can take a minutes. The Load Balancer has health-checks that will pull the instances
-  #   out of rotation and those checks can be directed at either varnish itself, the services through
-  #   varnish, or the services directly. So, changing the order here means evalulating whether or not
-  #   the ELB health-checks make sense to begin diverting traffic at the right time (vs. returning 503s)
+def deregister_frontend(name, host, log_path)
+  # Use the AWS-provided scripts to cleanly deregister a frontend instance from its load balancer(s),
+  # with zero downtime and support for auto-scaling groups.
   command = [
-    'sudo service varnish stop',
-    'sudo service dashboard stop',
-    'sudo service pegasus stop',
+    "cd #{rack_env}",
+    'bin/deploy_frontend/deregister_from_elb.sh',
```

Comment:
Is it possible for deregistration to fail? If it fails, the whole deploy process should halt